src/main/java/com/spikes2212/control/PIDFSettings.java

```diff
@@ -0,0 +1,122 @@
+package com.spikes2212.control;
+
+import java.util.function.Supplier;
+
+/**
+ * The PID constants used in a PID loop, and the error tolerance and loop wait time.
+ */
+public class PIDFSettings {
+    /**
+     * The proportional component of the PIDF settings.
+     */
+    private Supplier<Double> kP;
+
+    /**
+     * The integral component of the PIDF settings.
+     */
+    private Supplier<Double> kI;
+
+    /**
+     * The derivative component of the PIDF settings.
+     */
+    private Supplier<Double> kD;
+
+    /**
+     * The feed forward component of the PIDF settings required to stay on target.
+     */
+    private Supplier<Double> kF;
+
+    /**
+     * The acceptable distance from the target.
```

Comment:
the acceptable distance from the `setpoint`, the `target` is an area that includes the setpoint with a  margin of `tolerance` in both directions